sky/backends/backend_utils.py

```diff
@@ -751,70 +628,38 @@ def remove_cluster(
-            return
-
-        # Scan for end of previous config.
-        cursor = start_line_idx
-        while cursor > 0 and len(config[cursor].strip()) > 0:
-            cursor -= 1
-        prev_end_line_idx = cursor
-
-        # Scan for end of the cluster config.
-        end_line_idx = None
-        cursor = start_line_idx + 1
-        start_line_idx -= 1  # remove auto-generated comment
-        while cursor < len(config):
-            if config[cursor].strip().startswith(
-                    '# ') or config[cursor].strip().startswith('Host '):
-                end_line_idx = cursor
-                break
-            cursor += 1
-
-        # Remove sky-generated config and update the file.
-        config[prev_end_line_idx:end_line_idx] = [
-            '\n'
-        ] if end_line_idx is not None else []
-        with open(config_path, 'w') as f:
-            f.write(''.join(config).strip())
-            f.write('\n' * 2)
-
-        SSHConfigHelper._remove_multinode_config(cluster_name)
-
-    @classmethod
-    def _remove_multinode_config(
-        cls,
-        cluster_name: str,
-    ):
-        config_path = os.path.expanduser(cls.ssh_conf_path)
-        if not os.path.exists(config_path):
-            return
-
-        extra_path_name = cls.ssh_multinode_path.format(cluster_name)
-        extra_config_path = os.path.expanduser(extra_path_name)
-        common_utils.remove_file_if_exists(extra_config_path)
+        # Ensures backward compatibility
+        if start_line_idx is not None:
+
+            # Scan for end of previous config.
+            cursor = start_line_idx
+            while cursor > 0 and len(config[cursor].strip()) > 0:
+                cursor -= 1
+            prev_end_line_idx = cursor
+
+            # Scan for end of the cluster config.
+            end_line_idx = None
+            cursor = start_line_idx + 1
+            start_line_idx -= 1  # remove auto-generated comment
+            while cursor < len(config):
+                if config[cursor].strip().startswith(
+                        '# ') or config[cursor].strip().startswith('Host '):
+                    end_line_idx = cursor
+                    break
+                cursor += 1
 
-        # Delete include statement
-        sky_autogen_comment = ('# Added by sky (use `sky stop/down '
-                               f'{cluster_name}` to remove)')
-        with open(config_path) as f:
-            config = f.readlines()
+            # Remove sky-generated config and update the file.
+            config[prev_end_line_idx:end_line_idx] = [
+                '\n'
+            ] if end_line_idx is not None else []
+            with open(config_path, 'w') as f:
+                f.write(''.join(config).strip())
+                f.write('\n' * 2)
 
-        for i, line in enumerate(config):
-            config_str = line.strip()
-            if f'Include {extra_config_path}' in config_str:
-                with open(config_path, 'w') as f:
```

Comment:
We should remove `Include` lines when the cluster is termianted?

Reply:
Okay so the idea is that we are including all the configs in the ssh_dir. So the added Include statement will be like ~
`Include ~/.sky/generated/ssh/*` (for a UNIX based system), so I thought we should just have it there forever as it does not break any logic regardless. Do you think we want to remove this once all the clusters are terminated?

Reply:
Good point! One thing to confirm: will the `ssh` work for user's own ssh config, when the `~/.sky/generated/ssh` folder does not exist and `Include ~/.sky/generated/ssh/*`? Just to make sure there is no corner case.

Reply:
So the `~/.sky/generated/ssh` has to exist, since whenever we are adding a new cluster it is going to created. And we only will have to remove the clusters from the config when remove is called which is handled. One thing that you mention which is lacking here now that I realise is that we can actually just go ahead and remove the old include statement if it was added for any specific multinode cluster to maintain a clean config. I will do that in the new commit. Maybe in the future we can deprecate it.